### PR TITLE
[N8N-4399]: Audit the use of v-html

### PR DESCRIPTION
**Pull request by OlegIvaniv**
[N8N-4399](https://linear.app/n8n/issue/N8N-4399)

I've gone over the usage of `v-html` in the editor-ui package. The feasible occurrences got replaced with component interpolation syntax. In some cases `v-html` was not required at all. And finally, I've stumbled across the pattern we use where we'd use `v-html` to render a list of items with a title(src/components/Node.vue#9, src/components/NodeCredentials.vue#45, src/components/ParameterIssues.vue#4) so I've created a new component to externalise this use-case — TitledList. 

Below I'm attaching the audit list of all the occurrences of `v-html` with some notes on handling them.
```html
40 results - 25 files

<!-- `infoTextErrorMessage` would be http response error which shouldn't contain HTML so we can use v-text -->
<!-- } catch(error) {
      if(error.httpStatusCode && error.httpStatusCode === 400) {
        this.infoTextErrorMessage = error.message;
      } else {
        this.$showError(
          error,
          this.$locale.baseText('settings.communityNodes.messages.install.error'),
        );
      }
    } -->
src/components/CommunityPackageInstallModal.vue:
  52  						:class="[$style.infoText, infoTextErrorMessage ? $style.error : '']"
  53: 						v-html="infoTextErrorMessage"
  54  					></span>


<!-- Description is just a simple text locale, it's safe to change it to v-text -->
<!-- "settings.communityNodes.confirmModal.update.description": "We recommend you deactivate workflows that use any of the package's nodes and reactivate them once the update is completed"-->
src/components/CommunityPackageManageConfirmModal.vue:
  15  					<template>
  16: 						<span v-html="getModalContent.description"></span>
  17  					</template>


<!-- Credential description might contain html with a link to the docs -->
<!-- credType example: {
  "displayName": "Subdomain",
  "name": "subdomain",
  "type": "string",
  "default": "",
  "placeholder": "n8n",
  "description": "Subdomain of cloud-hosted ERPNext instance. For example, \"n8n\" is the subdomain in: <code>https://n8n.erpnext.com</code>",
  "displayOptions": {
    "show": {
      "environment": ["cloudHosted"]
    }
  }
} -->
src/components/CredentialsSelect.vue:
  27  							class="option-description"
  28: 							v-html="credType.description"
  29  						/>

<!-- One of the return branches of `statusTooltipText` includes a break so it's not suitable for interpolation -->
<!-- "executionsList.statusTooltipText.theWorkflowExecutionWasARetryOfAndFailed": "The workflow execution was a retry of {entryRetryOf} and failed.<br />New retries have to be started from the original execution.", -->
src/components/ExecutionsList.vue:
  83  						<n8n-tooltip placement="top" >
  84: 							<div slot="content" v-html="statusTooltipText(scope.row)"></div>
  85  							<span class="status-badge running" v-if="scope.row.waitTill">


<!-- Coming soon feature description contains href with a link to learn more about the feature, since it's dynamical it's not suitable for interpolation -->
<!-- 	"fakeDoor.settings.logging.infoText": "You can already write logs to a file or the console using environment variables. <a href=\"https://docs.n8n.io/hosting/logging/\" target=\"_blank\">More info</a>", -->
src/components/FeatureComingSoon.vue:
  8  					<template>
  9: 						<span v-html="$locale.baseText(featureInfo.infoText)"></span>
  10  					</template>

<!-- Contains only line-break, not suitable for interpolation -->
<!-- "dataMapping.dragFromPreviousHint": "Map data from previous nodes to <b>{name}</b><br/> by first clicking this button", -->
src/components/InputPanel.vue:
  38  				<n8n-tooltip v-if="!readOnly" :manual="true" :value="showDraggableHint && showDraggableHintWithDelay">
  39: 					<div slot="content" v-html="$locale.baseText('dataMapping.dragFromPreviousHint',  { interpolate: { name: focusedMappableInput } })"></div>
  40  					<NodeExecuteButton type="secondary" :transparent="true" :nodeName="currentNodeName" :label="$locale.baseText('ndv.input.noOutputData.executePrevious')" @execute="onNodeExecute" telemetrySource="inputs" />


src/components/Node.vue:
   8  						<n8n-tooltip placement="bottom" >
   <!-- Refactored to use regular HTML syntax via TitledList component -->
   9: 							<div slot="content" v-html="nodeIssues"></div>
  10  							<font-awesome-icon icon="exclamation-triangle" />

  14  						<n8n-tooltip placement="bottom">
  <!-- This could be two different locales, none of them contain html so it should be safe to use v-text -->
  <!-- "node.nodeIsWaitingTill": "Node is waiting until {date} {time}",
	"node.theNodeIsWaitingIndefinitelyForAnIncomingWebhookCall": "The node is waiting for an incoming webhook call (indefinitely)", -->
  15: 							<div slot="content" v-html="waiting"></div>
  16  							<font-awesome-icon icon="clock" />

<!-- Refactored to use regular HTML syntax via TitledList component -->
src/components/NodeCredentials.vue:
  44  						<n8n-tooltip placement="top" >
  45: 							<div slot="content" v-html="`${$locale.baseText('nodeCredentials.issues')}:<br />&nbsp;&nbsp;- ` + issues.join('<br />&nbsp;&nbsp;- ')"></div>
  46  							<font-awesome-icon icon="exclamation-triangle" />


<!-- Refactored to use component interpolation -->
src/components/NodeSettings.vue:
  31  					<span
  32: 						v-html="$locale.baseText('nodeSettings.communityNodeUnknown.description', { interpolate: { packageName: node.type.split('.')[0] } })"
  33  						@click="onMissingNodeTextClick"

<!-- Refactored to use component interpolation -->
  44  			<span v-else
  45: 				v-html="
  46  					$locale.baseText('nodeSettings.nodeTypeUnknown.description',

<!-- ndv.output.staleDataWarning.regular is using new line break, not suitable for component interpolation  -->
src/components/OutputPanel.vue:
  36  					<template>
  37: 						<span v-html="$locale.baseText(
  38  							hasPinData ? 'ndv.output.staleDataWarning.pinData' : 'ndv.output.staleDataWarning.regular'

<!-- Using custom translation output method, not suitable for component interpolation -->
src/components/ParameterInput.vue:
  211  							class="option-description"
  212: 							v-html="getOptionsOptionDescription(option)"
  213  						></div>

<!-- Using custom translation output method, not suitable for component interpolation -->
  244  							class="option-description"
  245: 							v-html="getOptionsOptionDescription(option)"
  246  						></div>

<!-- It should be safe to turn this into v-html since all the .hint locales are simple text -->
<!-- 
    n8n/packages/editor-ui/src/plugins/i18n/locales/en.json
    356,25: 	"ndv.input.noOutputData.hint": "(From the earliest node that has no output data yet)",
    678,18: 	"runData.linking.hint": "Link displayed input and output runs",
    679,20: 	"runData.unlinking.hint": "Unlink displayed input and output runs",
 -->
src/components/ParameterInputHint.vue:
  3  		<n8n-text size="xsmall" color="text-base" v-if="hint">
  4: 			<div ref="hint" v-html="hint"></div>
  5  		</n8n-text>

<!-- Refactored to use regular HTML syntax via TitledList component -->
src/components/ParameterIssues.vue:
  3  		<n8n-tooltip placement="top" >
  4: 			<div slot="content" v-html="`${$locale.baseText('parameterInput.issues')}:<br />&nbsp;&nbsp;- ` + issues.join('<br />&nbsp;&nbsp;- ')"></div>
  5  			<font-awesome-icon icon="exclamation-triangle" />

<!-- Line break, not suitable -->
<!-- 671,3: 	"pushConnectionTracker.cannotConnectToServer": "You have a connection issue or the server is down. <br />n8n should reconnect automatically once the issue is resolved.", -->
src/components/PushConnectionTracker.vue:
  4  			<n8n-tooltip placement="bottom-end" >
  5: 				<div slot="content" v-html="$locale.baseText('pushConnectionTracker.cannotConnectToServer')"></div>
  6  				<span>

<!-- Line break, not suitable -->
<!-- "ndv.output.tooMuchData.message": "The node contains {size} MB of data. Displaying it may cause problems. <br /> If you do decide to display it, avoid the JSON view.", -->
src/components/RunData.vue:
  217  				<n8n-text :bold="true" color="text-dark" size="large">{{ tooMuchDataTitle }}</n8n-text>
  218: 				<n8n-text align="center" tag="div"><span v-html="$locale.baseText('ndv.output.tooMuchData.message', { interpolate: {size: dataSizeInMB }})"></span></n8n-text>
  219  

<!-- Refactored and separated out the <img> element from translation -->
<!-- "dataMapping.dragColumnToFieldHint": "<img src='/static/data-mapping-gif.gif'/> Drag onto a field to map column to that field", -->
src/components/RunDataTable.vue:
  17  						<n8n-tooltip placement="bottom-start" :disabled="!mappingEnabled || showHintWithDelay" :open-delay="1000">
  18: 							<div slot="content" v-html="$locale.baseText('dataMapping.dragColumnToFieldHint')"></div>
  19  							<Draggable type="mapping" :data="getExpression(column)" :disabled="!mappingEnabled" @dragstart="onDragStart" @dragend="(column) => onDragEnd(column)">

  35  										<n8n-tooltip v-if="mappingEnabled" placement="bottom-start" :manual="true" :value="i === 0 && showHintWithDelay">
  36: 											<div v-if="focusedMappableInput" slot="content" v-html="$locale.baseText('dataMapping.tableHint', { interpolate: { name: focusedMappableInput } })"></div>
<!-- Refactored and separated out the <img> element from translation -->
<!-- "dataMapping.dragColumnToFieldHint": "<img src='/static/data-mapping-gif.gif'/> Drag onto a field to map column to that field", -->
  37: 											<div v-else slot="content" v-html="$locale.baseText('dataMapping.dragColumnToFieldHint')"></div>
  38  											<div :class="$style.dragButton">

<!-- Subheader locale is a simple text, no need for v-htmll -->
<!-- subheader(): string {
			const serviceName = this.nodeType ? getTriggerNodeServiceName(this.nodeType) : '';
			if (this.isActivelyPolling) {
				return this.$locale.baseText('ndv.trigger.pollingNode.fetchingHint', {
					interpolate: { name: serviceName },
				});
			}

			return '';
		}, -->
src/components/TriggerPanel.vue:
  57  						<n8n-text v-if="subheader">
  58: 							<span v-html="subheader"></span>
  59  						</n8n-text>
<!-- activationHint is highly dynamic locale which is some cases might include HTML -->
  71  				<n8n-text size="small" @click="onLinkClick" v-if="activationHint">
  72: 					<span v-html="activationHint"></span>&nbsp;
  73  				</n8n-text>

<!-- Line break, not suitable -->
<!-- "versionCard.thisVersionHasASecurityIssue": "This version has a security issue.<br />It is listed here for completeness.", -->
src/components/VersionCard.vue:
  10  					<template>
  11: 						<span v-html="$locale.baseText('versionCard.thisVersionHasASecurityIssue')"></span>
  12  					</template>

<!-- Version might sometime include HTML, not suitable for component interpolation -->
<!-- Versions details are served from https://api.n8n.io/api/versions/ which comes from Strapi CMS.
  This could potentionaly be dangerous if somebody would get acces to it. IMHO, it would be better to serve just static text or custom content objects.
-->
  30  		<div>
  31: 			<div v-if="version.description" v-html="version.description" :class="$style.description"></div>
  32  			<div v-if="version.nodes && version.nodes.length > 0" :class="$style.nodes">

<!-- Line break, not suitable -->
<!--  964,3: 	"workflowActivator.theWorkflowIsSetToBeActiveBut": "The workflow is activated but could not be started.<br />Click to display error message." -->
src/components/WorkflowActivator.vue:
  18  			<n8n-tooltip placement="top">
  19: 				<div @click="displayActivationError" slot="content" v-html="$locale.baseText('workflowActivator.theWorkflowIsSetToBeActiveBut')"></div>
  20  				<font-awesome-icon @click="displayActivationError" icon="exclamation-triangle" />


src/components/WorkflowSettings.vue:
<!-- Line break, not suitable -->
<!-- 998,3: 	"workflowSettings.helpTexts.errorWorkflow": "A second workflow to run if the current one fails.<br />The second workflow should an 'Error Trigger' node." -->
   15  						<n8n-tooltip class="setting-info" placement="top" >
   16: 							<div slot="content" v-html="helpTexts.errorWorkflow"></div>
   17  							<font-awesome-icon icon="question-circle" />
<!-- Simple text, could use v-text -->
   34  						<n8n-tooltip class="setting-info" placement="top" >
   35: 							<div slot="content" v-html="helpTexts.timezone"></div>
   36  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
   53  						<n8n-tooltip class="setting-info" placement="top" >
   54: 							<div slot="content" v-html="helpTexts.saveDataErrorExecution"></div>
   55  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
   72  						<n8n-tooltip class="setting-info" placement="top" >
   73: 							<div slot="content" v-html="helpTexts.saveDataSuccessExecution"></div>
   74  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
   91  						<n8n-tooltip class="setting-info" placement="top" >
   92: 							<div slot="content" v-html="helpTexts.saveManualExecutions"></div>
   93  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
  110  						<n8n-tooltip class="setting-info" placement="top" >
  111: 							<div slot="content" v-html="helpTexts.saveExecutionProgress"></div>
  112  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
  129  						<n8n-tooltip class="setting-info" placement="top" >
  130: 							<div slot="content" v-html="helpTexts.executionTimeoutToggle"></div>
  131  							<font-awesome-icon icon="question-circle" />

<!-- Simple text, could use v-text -->
  144  							<n8n-tooltip class="setting-info" placement="top" >
  145: 								<div slot="content" v-html="helpTexts.executionTimeout"></div>
  146  								<font-awesome-icon icon="question-circle" />

<!-- Line break, not suitable -->
<!--  190,3: 	"executionDetails.readOnly.youreViewingTheLogOf": "You're viewing the log of a previous execution. You cannot<br />\n\t\tmake changes since this execution already occurred. Make changes<br />\n\t\tto this workflow by clicking on its name on the left."-->
src/components/MainHeader/ExecutionDetails/ReadOnly.vue:
  3  		<div slot="content">
  4: 			<span v-html="$locale.baseText('executionDetails.readOnly.youreViewingTheLogOf')"></span>
  5  		</div>

<!-- Simple text, could use v-text -->
  7  			<font-awesome-icon icon="exclamation-triangle" />
  8: 			<span v-html="$locale.baseText('executionDetails.readOnly.readOnly')"></span>
  9  		</div>

<!-- This would be possible to refactor to component interpolation but we can't. Because `generic.communityNode.tooltip` locale is used also in NodeSettingsTabs where we populate options array dynamically -->
<!-- 
  options.push({
    icon: 'cube',
    value: 'communityNode',
    align: 'right',
    tooltip: this.$locale.baseText('generic.communityNode.tooltip', {
      interpolate: {
        docUrl: COMMUNITY_NODES_INSTALLATION_DOCS_URL,
        packageName: this.packageName,
      },
    }),
  });
 -->
src/components/NodeCreator/NodeItem.vue:
  24  						slot="content"
  25: 						v-html="$locale.baseText('generic.communityNode.tooltip', { interpolate: { packageName: nodeType.name.split('.')[0], docURL: COMMUNITY_NODES_INSTALLATION_DOCS_URL } })"
  26  						@click="onCommunityNodeTooltipClick"

<!-- Refactored to use component interpolation -->
src/views/SettingsApiView.vue:
  15  					<n8n-info-tip :bold="false">
  16: 						<span v-html="$locale.baseText('settings.api.view.info')"></span>
  17  					</n8n-info-tip>

<!-- Refactored to component interpolation -->
src/views/SettingsUsersView.vue:
  7  						<n8n-tooltip :disabled="isSmtpSetup" placement="bottom">
  8: 							<div slot="content" v-html="$locale.baseText('settings.users.setupSMTPToInviteUsers')"></div>
  9  							<div>

<!-- Not suitable — one of the return branches for `endOfSearchMessage` computed contains html local -->
<!-- "templates.endResult": "Share your own useful workflows through your <a href='https://n8n.io/dashboard' target='_blank'>n8n.io account</a>", -->
src/views/TemplatesSearchView.vue:
  65  						<n8n-text size="medium" color="text-base">
  66: 							<span v-html="endOfSearchMessage" />
  67  						</n8n-text>

```

### CODE REVIEW

File: packages/editor-ui/src/components/NodeSettings.vue [hunk 1]
```diff
@@ -28,11 +28,14 @@
...
-					>
-					</span>
+					<i18n path="nodeSettings.communityNodeUnknown.description" tag="span" @click="onMissingNodeTextClick">
+						<template #action>
+							<a
+								:href="`https://www.npmjs.com/package/${node.type.split('.')[0]}`"
+								target="_blank"
+							>{{ node.type.split('.')[0] }}</a>
+						</template>
+					</i18n>
```
Comment (MiloradFilipovic) [inline packages/editor-ui/src/components/NodeSettings.vue:38]: Look like unknown community mode modal is not rendering correctly with these changes (it shows additional content after _how to install_ link):
SCR-20220905-fgt

**Review by MiloradFilipovic — DISMISSED** [review]
Nice work @OlegIvaniv, i left one thing that needs to be addressed. Once that is done, feel free to merge.

**Review by MiloradFilipovic — APPROVED** [review]
Aproving after fixing and resolving conflicts

**Comment by janober** [comment]
Got released with `n8n@0.193.4`

**Pull request merged**